src/bin/dependency_check.rs

```diff
@@ -0,0 +1,223 @@
+    check_ffmpeg_command(required_ffmpeg_version)
+}
+
+#[cfg(target_os = "macos")]
+fn check_ffmpeg(required_ffmpeg_version: &str) -> Result<bool> {
+    let command_result = check_ffmpeg_command(required_ffmpeg_version)?;
+    if !command_result {
+        info!("Checking if ffmpeg is installed as homebrew keg-only");
+        return check_ffmpeg_homebrew(required_ffmpeg_version);
+    }
+    Ok(command_result)
+}
+
+fn check_ffmpeg_command(required_ffmpeg_version: &str) -> Result<bool> {
+    let ffmpeg_result = Command::new("ffmpeg").arg("-version").output();
+    match ffmpeg_result {
+        Ok(ffmpeg_output) => {
+            let ffmpeg_output = String::from_utf8(ffmpeg_output.stdout)?.trim().to_string();
+            match match_ffmpeg_version(&ffmpeg_output) {
+                Some(version) if version == required_ffmpeg_version => Ok(true),
+                Some(version) => {
+                    warn!(
+                        installed_ffmpeg_version = version,
+                        required_ffmpeg_version,
+                        "Inatelled version doesn't match the required version."
+                    );
+                    Ok(false)
+                }
+                None => {
+                    warn!("Failed to parse FFmpeg version.");
+                    Ok(false)
+                }
+            }
+        }
+        Err(_) => {
+            warn!("Failed to run FFmpeg.");
+            Ok(false)
+        }
+    }
+}
+
+fn match_ffmpeg_version(ffmpeg_output: &str) -> Option<&str> {
+    let re = Regex::new(r"(?m)^ffmpeg version \D*(\d+\.\d+)")
+        .expect("Failed to compile regular expression");
+    let caps = re.captures(&ffmpeg_output);
+    match caps {
+        Some(caps) => caps.get(1).map(|cap| cap.as_str()),
+        None => {
+            warn!("Failed to parse FFmpeg version.");
+            None
+        }
+    }
+}
+
+#[cfg(target_os = "macos")]
+fn check_ffmpeg_homebrew(required_ffmpeg_version: &str) -> Result<bool> {
+    let required_ffmpeg_version_brew =
+        &required_ffmpeg_version[..required_ffmpeg_version.find(".").unwrap_or(1)];
+    let brew_output = Command::new("brew").arg("list").output()?;
+    let brew_output = String::from_utf8(brew_output.stdout)?.trim().to_string();
+
+    let ffmpeg_string = format!("ffmpeg@{required_ffmpeg_version_brew}");
+
+    let re = Regex::new(&format!("(?m){ffmpeg_string}"))?;
+    if re.is_match(&brew_output) {
+        Ok(true)
+    } else {
+        warn!("FFmpeg installation not found in homebrew");
+        Ok(false)
+    }
+}
+
+#[cfg(test)]
+mod dependency_check_test {
+    use super::*;
+    use std::{fs, path::PathBuf};
+
+    // TODO: (@jbrs) test for linux output
+    #[test]
+    fn ffmpeg_macos_regex_test() {
+        let crate_root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let ffmpeg_output =
+            fs::read_to_string(crate_root.join("src/bin/ffmpeg_output_macos.txt")).unwrap();
```

Comment:
as mentioned bellow I suggest inlining text here, however for similiar cases in tests you would usually use `include_str!` macro to read file content during compilation